clisops/core/subset.py

```diff
@@ -771,34 +771,52 @@ def shape_bbox_indexer(ds, poly):
     -----
     This is used in particular to restrict the domain of a dataset before computing the weights for a spatial average.
     """
-    rectilinear = is_rectilinear(ds)
-
-    # Union of all geometries -> shapely.geometry.Polygon
-    geom = poly.geometry.unary_union
-
-    # Shape envelope
-    if rectilinear:
-        envelope = geom.minimum_rotated_rectangle
-    else:
-        # For curvilinear grids, the convex hull seems safer than the rotated_rectangle.
-        envelope = geom.convex_hull
+    # Cling wrap around shapes: GeoSeries -> Polygon
+    # The first `convex_hull` is necessary to remove any holes in the polygon.
+    # The `GeoSeries.unary_union` is necessary to merge all shapes into a single polygon.
+
+    if isinstance(poly, gpd.GeoDataFrame):
+        # Note that this function is somewhat redundant with functionality found in rioxarray (see `clip` and `Window`).
+        geom = poly.geometry
+        hull = geom.convex_hull.unary_union.convex_hull
+    elif isinstance(poly, gpd.array.GeometryArray):
+        hull = poly.convex_hull.unary_union().convex_hull
+    elif isinstance(poly, (list, tuple)):
+        hpoly = [p.convex_hull for p in poly]
+        hull = unary_union(hpoly).convex_hull
 
     # If polygon sits on the grid boundary, we need to roll the grid's coordinates and this is not supported.
-    if not grid_exterior_polygon(ds).contains(envelope):
+    if not grid_exterior_polygon(ds).contains(hull):
         return {}
 
-    # Create index from edge vertices (last item is just a copy of the first to close the polygon)
-    elon, elat = map(np.array, zip(*envelope.boundary.coords[:-1]))
+    # If the grid is rectilinear, we can use the minimum rotated rectangle to simplify the outline further.
+    rectilinear = is_rectilinear(ds)
+    if rectilinear:
+        hull = hull.minimum_rotated_rectangle
+
+    if hasattr(hull.boundary, "geoms"):
+        # Handle cases with 1 point. Putting the same point twice to avoid issues with zip later.
+        if len(hull.boundary.geoms) == 0:
+            coords = hull.xy, hull.xy
+        # Handle cases with 2 points
+        elif len(hull.boundary.geoms) <= 2:
```

Comment:
What happens with `len(...) > 2` ?

Reply:
Then it's a polygon and it should be handled by the else below. But I see your point, I could replace the elif by an else.